### PR TITLE
Add more CI targets for cross compilation

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**
Continuing of #952.

### CODE REVIEW

File: ci/azure-cross-compile.yml [hunk 1]
```diff
@@ -34,14 +38,6 @@ jobs:
         vmImage: ubuntu-16.04
         target: x86_64-sun-solaris
 
-      Linux_ARM:
-        vmImage: ubuntu-16.04
-        target: arm-linux-androideabi
```
Comment (Thomasdezeeuw) [inline ci/azure-cross-compile.yml]: This was the same as the `Android` target (see above) so I removed it.

File: ci/azure-cross-compile.yml [hunk 2]
```diff
@@ -37,7 +53,7 @@ jobs:
     - script: rustup target add $(target)
       displayName: "Add target"
 
-    - script: cargo check --target $(target)
+    - script: cargo build --target $(target)
```
Comment (asomers) [inline ci/azure-cross-compile.yml]: Why change `cargo check` to `cargo build`?  Won't that just slow things down for little benefit?

Comment (Thomasdezeeuw) [reply ci/azure-cross-compile.yml]: That was in the original pr, see https://github.com/tokio-rs/mio/pull/952/files#diff-424edf9074d0455636ef5bf26a0c9cd4R56. @carllerche any reason why you changed it?

Comment (carllerche) [reply ci/azure-cross-compile.yml]: I'm fine either way. So up to y'all.

Comment (carllerche) [reply ci/azure-cross-compile.yml]: I don't remember why I changed it

Comment (Thomasdezeeuw) [reply ci/azure-cross-compile.yml]: I'll change it back, because I do think its faster.

**Pull request merged**